test/e2e/operator_test.go

```diff
@@ -77,47 +82,87 @@ func newIngressController(name, ns, domain string, epType operatorv1.EndpointPub
 	}
 }
 
-func TestOperatorAvailable(t *testing.T) {
-	cl, _, err := getClient()
-	if err != nil {
-		t.Fatal(err)
+func clusterOperatorConditionMap(conditions ...configv1.ClusterOperatorStatusCondition) map[string]string {
+	conds := map[string]string{}
+	for _, cond := range conditions {
+		conds[string(cond.Type)] = string(cond.Status)
+	}
+	return conds
+}
+
+func operatorConditionMap(conditions ...operatorv1.OperatorCondition) map[string]string {
+	conds := map[string]string{}
+	for _, cond := range conditions {
+		conds[string(cond.Type)] = string(cond.Status)
 	}
+	return conds
+}
+
+func conditionsMatchExpected(expected, actual map[string]string) bool {
+	filtered := map[string]string{}
+	for k := range actual {
+		if _, comparable := expected[k]; comparable {
+			filtered[k] = actual[k]
+		}
+	}
+	return reflect.DeepEqual(expected, filtered)
+}
 
+func waitForClusterOperatorConditions(cl client.Client, conditions ...configv1.ClusterOperatorStatusCondition) error {
 	co := &configv1.ClusterOperator{}
-	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+	return wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
 		if err := cl.Get(context.TODO(), types.NamespacedName{Name: ingresscontroller.IngressClusterOperatorName}, co); err != nil {
-			return false, nil
+			return false, err
 		}
 
-		for _, cond := range co.Status.Conditions {
-			if cond.Type == configv1.OperatorAvailable &&
-				cond.Status == configv1.ConditionTrue {
-				return true, nil
-			}
-		}
+		expected := clusterOperatorConditionMap(conditions...)
+		current := clusterOperatorConditionMap(co.Status.Conditions...)
+		return conditionsMatchExpected(expected, current), nil
+	})
+}
 
-		return false, nil
+func waitForIngressControllerCondition(cl client.Client, name types.NamespacedName, conditions ...operatorv1.OperatorCondition) error {
+	ic := &operatorv1.IngressController{}
+	return wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		if err := cl.Get(context.TODO(), name, ic); err != nil {
+			return false, err
+		}
+		expected := operatorConditionMap(conditions...)
+		current := operatorConditionMap(ic.Status.Conditions...)
+		return conditionsMatchExpected(expected, current), nil
 	})
+}
+
+func TestOperatorAvailable(t *testing.T) {
+	cl, _, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected := []configv1.ClusterOperatorStatusCondition{
+		{Type: configv1.OperatorAvailable, Status: configv1.ConditionTrue},
+	}
+	err = waitForClusterOperatorConditions(cl, expected...)
```

Comment:
Because you made these helpers variadic, building a slice is unnecessary—this code could be simplified as follows:
```go
		expected := configv1.ClusterOperatorStatusCondition{
			Type: configv1.OperatorAvailable,
			Status: configv1.ConditionTrue,
		}
		err = waitForClusterOperatorConditions(cl, expected)
```
Alternatively, if nothing is taking advantage of the variadic parameters, is there any reason not to use slice-valued parameters instead of variadic parameters?